tests/unit/model/tracked-array-test.js

```diff
@@ -7,297 +7,336 @@ import DefaultSchema from 'ember-m3/services/m3-schema';
 import { A } from '@ember/array';
 import ManagedArray from 'ember-m3/managed-array';
 
-module('unit/model/tracked-array', function(hooks) {
-  setupTest(hooks);
-
-  hooks.beforeEach(function() {
-    this.sinon = sinon.createSandbox();
-    this.owner.register(
-      'service:m3-schema',
-      class TestSchema extends DefaultSchema {
-        includesModel(modelName) {
-          return /^com.example.bookstore\./i.test(modelName);
-        }
-
-        computeNestedModel(key, value /*, modelName, schemaInterface */) {
-          if (typeof value === 'object' && value !== null && !Array.isArray(value)) {
-            return {
-              attributes: value,
-            };
-          }
+function computeNestedModel(key, value /*, modelName, schemaInterface */) {
+  if (typeof value === 'object' && value !== null && !Array.isArray(value)) {
```

Comment:
we do effectively this check a lot of places in both our code and our tests, and it seems slightly different every time. This is likely to be a hot path in our actual code, so we should audit at some point to ensure that the simplest bailout is being done prior to the typeof check.